sematic/resolvers/local_resolver.py

```diff
@@ -182,14 +204,28 @@ def _notify_pipeline_update(self):
         root_future = self._futures[0]
         api_client.notify_pipeline_update(self._runs[root_future.id].calculator_path)
 
-    def _resolution_did_succeed(self) -> None:
-        super()._resolution_did_succeed()
+    def _resolution_did_succeed(self, root_future: AbstractFuture) -> None:
+        super()._resolution_did_succeed(root_future)
+        self._update_resolution_status(root_future.id, ResolutionStatus.COMPLETE)
         self._notify_pipeline_update()
 
-    def _resolution_did_fail(self) -> None:
-        super()._resolution_did_fail()
+    def _resolution_did_fail(
+        self, root_future: AbstractFuture, due_to_calculator_error: bool
```

Comment:
Not a big fan of adding `due_to_calculator_error` here. It breaks symmetry with other hooks.

Is there any other way to pass that info?
Or should we simply keep symmetry between the hook name and the status? By that I mean `_resolution_did_fail` is only called when the resolution actually failed (i.e. `ResolutionStatus.FAILED`) otherwise `_resolution_did_succeed` is called (regardless of user code errors).

Reply:
How about we pass in the exception to the hook. There really is extra information in the exception that resolvers might want to handle differently. No reason to be symmetric about that: other hooks wouldn't expect an exception arg because they are not about failures.

Reply:
Yes to passing the exception instead of `due_to_calculator_error`. Passing the exception is more generic and as you said can be used for other purposes.